integration-tests/connect_test.go

```diff
@@ -45,6 +46,34 @@ func TestGoCqlConnect(t *testing.T) {
 	require.Equal(t, "fake", iter.Columns()[0].Name)
 }
 
+func TestProtocolVersionNegotiation(t *testing.T) {
+	c := setup.NewTestConfig("", "")
```

Comment:
we're still missing the multiple test cases that I requested. Feel free to look at other tests that contain "sub tests". We should do the same here with a matrix of C* versions and protocol versions and expected outcomes.